src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerAncestorFingerprintFacet.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.annotation.Nonnull;
+
+/**
+ * Facet representing the fact that some Docker image was derived from this one.
+ * This facet would be added to the fingerprint corresponding to the {@code FROM} instruction in a {@code Dockerfile}.
+ * {@link #getDescendantImageId} indicates the image that was built from that file.
+ * @see DockerFingerprints#addFromFacet
+ * @see DockerDescendantFingerprintFacet
+ */
+public class DockerAncestorFingerprintFacet extends DockerRunPtrFingerprintFacet {
```

Comment:
I don't get why this is the Ancestor Facet that points to a descendant
(and the Descendant facet points to the Ancestor).

i.e. Why doesn't the descendant Facet point to the descendant image and the Ancestor facet point to the Ancestor?